Add: new meeting notes April 23 (#141)
datproject/comm-comm: meeting-notes/49-23Apr2020.md

# dat protocol Community communication

23 Apr 2020

## People

* Diego (@dpaez)
* mauve (@rangermauve)
* Kevin Faaborg (@zootela)
* Peg

## Meeting Notes

### Intro

> - mauve: working with datdot and ... .Release a couple of projects recently (https://github.com/RangerMauve/p2plex/). Multiplex peers streams. Chat, apps, and different things going over a single connection. More easily to dedup connections. Similar idea to libp2p: combine the discovery and encryption and multiplex. Initially done for the datdot project. [dat gossip](https://github.com/RangerMauve/dat-gossip) like a PoC of gossiping keys, eg chat app sharing hypercores, you can use this gossip module for share these hypercores and easily discover data.

Reviewer: ```suggestion
- mauve: working with datdot and ... .Release a couple of projects recently (https://github.com/RangerMauve/p2plex/). Multiplex peers streams. real-time communicaton apps, and different things going over a single connection. More easily to dedup connections. Similar idea to libp2p: combine the hyperswarm for peer discovery and noise for encryption / authentication and multiplex for sending multiple streams of data at once. Made for the datdot project. [dat gossip](https://github.com/RangerMauve/dat-gossip) like a PoC of gossiping keys, eg chat app sharing hypercores, you can use this gossip module for share these hypercore keys and easily discover data.
```

- https://gitlab.com/coboxcoop/core
- https://github.com/ameba23/metadb

## Event update

CFP coming soon, some dates are fixed.

We will probably end up using Big Blue Button since it gives a lot of flexibility.
We also want to explore more the idea around Mozilla Hubs, not only for presentations but also as a place for general conversation (side talks, chit chat)

### conflict resolution (cobox)

peg: read about bloom filtering to do this but decided to go with vector clocks instead of bloom clocks. A bit hard to test. kind of depend how people uses it. They are not  using actively working tasks. Different approach.
mauve: where do you store vector clocks?
peg: using a different feed (vector feed). frando proposes using more the metadata feed.
mauve: one vector clock or more like a profile thing?
peg: profile (multiple vectors?)
mauve: was looking around for different approaches.

peg: https://gitlab.com/coboxcoop/kappa-drive/-/blob/master/index.js#L502
https://gitlab.com/coboxcoop/kappa-drive/-/blob/master/index.js#L383

mauve: why store it in kappa instead of metadata?
peg: it is in the feed and the feed is aggregated using kappa core.

> mauve: trade off?

Reviewer: ```suggestion
mauve: is the tradeoff storing a bit of stuff in leveldb for improved reads?
```

peg: it was the way we see it.

### static site gens

kevin: every website starts with an index.html and that's it. You can upload and even edited from the browser. That's the feeling that I'm looking for.

> This can work on beaker and everywhere actually.

Reviewer: ```suggestion
This can work on beaker and everywhere actually, like [zeronet](https://zeronet.io/)
```
